### PR TITLE
Json parser

### LINKED ISSUE #6: Support AWS WAF logs stored in S3
AWS currently offers a method to send logs to ElasticSearch using Kinesis Firehose but it only supports public AWS ElasticSearch domain. No option to send the logs to ES in VPC or self hosted.

Do you think it will be possible to support these logs too? Didn't find any reliable solution to send WAF logs to ES; logstash still consumes too many resources for this task.

Thanks for this great project!

### PR DESCRIPTION
**Pull request by mpucholblasco**
# Notes
WAF log parser inspired on global JSON log parser.

# Issues related
Closes #6 .

### CODE REVIEW

File: logparser/jsonlogparser.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/common/jsontransform"
+)
+
+// JSONLogParserConfig JSONLogParser configuration
+type JSONLogParserConfig struct {
+	TimestampField string `config:"timestamp_field" validate:"required"`
+	TimestampKing  string `config:"timestamp_format" validate:"required"`
```
Comment (luismiramirez) [inline logparser/jsonlogparser.go]: `TimestampKind`?

File: logparser/kind.go [hunk 2]
```diff
@@ -152,96 +158,249 @@ func mustKindMapStringToType(o map[string]string) map[string]kindElement {
...
+			kindExtra: timeLayout,
+			name:      fmt.Sprintf("time layout (%s)", timeLayout),
+		}, nil
+	} else {
+		return kindElement{}, fmt.Errorf("Unsupported kind (%s)", v)
+	}
+}
+
+// parseToKind parses a value to convert it into the kind passed as argument
+// NOTE: tried to impre performance (obtained ~46.5ns/op) by using functions inside kindElement
```
Comment (d2bit) [inline logparser/kind.go]: impre -> improve

Comment (mpucholblasco) [reply logparser/kind.go]: fixed

File: logparser/jsonlogparser.go (hunk 1 above)
Comment (mpucholblasco) [reply logparser/jsonlogparser.go]: fixed

**Pull request merged**